vent/api/plugin_helpers.py

```diff
@@ -392,14 +411,41 @@ def prep_start(self,
                                                  branch,
                                                  version)
 
+            # look out for links to delete because they're defined externally
+            links_to_delete = set()
             # check and update links, volumes_from, network_mode
             for container in tool_d.keys():
                 if 'links' in tool_d[container]:
                     for link in tool_d[container]['links']:
-                        for c in tool_d.keys():
-                            if ('tmp_name' in tool_d[c] and
-                               tool_d[c]['tmp_name'] == link):
-                                tool_d[container]['links'][tool_d[c]['name']] = tool_d[container]['links'].pop(link)
+                        # add links to external services already running if necessary
+                        ext = 'external-services'
+                        if link in vent_config.options(ext)[1]:
+                            try:
+                                link_config = json.loads(vent_config.option(ext,
+                                                                            link)[1])
+                                ip_adr = link_config['ip_address']
+                                port = link_config['port']
+                                tool_d[container]['extra_hosts'] = {}
+                                # containers use lowercase names for
+                                # connections
+                                tool_d[container]['extra_hosts'][link.lower()] = ip_adr
+                                # create an environment variable for container
```

Comment:
this solution won't work, as it requires all plugins to now have to deal with the logic of whether to use external or the provided container.  that's something vent should handle and plugins shouldn't have to worry about. they should just expect that they can reach the service by name on the service's default port.